### PR TITLE
Task/Switch command tests

### PR DESCRIPTION
**Pull request by openbots-su**

### CODE REVIEW

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            _setVariableDefaultCase = new Variable.SetVariableCommand();
+
+            ScriptVariable caseVal = new ScriptVariable();
+            caseVal.VariableName = "case";
+            caseVal.VariableValue = caseString;
+            _engine.AutomationEngineContext.Variables.Add(caseVal);
+            ScriptVariable case1 = new ScriptVariable();
+            case1.VariableName = "caseOutput";
+            case1.VariableValue = "unassigned";
+            _engine.AutomationEngineContext.Variables.Add(case1);
```
Comment (openbots-ff) [inline OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs:47]: I'm having a little trouble following your naming conventions here. Could you make these ScriptVariable names either more descriptive, or add a comment explaining what each one is for?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs:35]: Addressed in 611e9ce

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// General Information about an assembly is controlled through the following
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("OpenBots.Commands.Switch.Test")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
+[assembly: AssemblyProduct("OpenBots.Commands.Switch.Test")]
+[assembly: AssemblyCopyright("Copyright ©  2021")]
```
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs:13]: Addressed in 611e9ce

File: OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs]: Addressed in 611e9ce
I plan to also include the task test in this PR once it is completed. I will also be updating old tests to use the new variable creation scheme.

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs (hunk 1 above)
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs]: Renamed in 824a3b0

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs [hunk 5]
```diff
@@ -37,25 +37,25 @@ public void ProcessesSwitch(string caseString)
...
 
-            ScriptVariable caseVal = new ScriptVariable();
-            caseVal.VariableName = "case";
-            caseVal.VariableValue = caseString;
-            _engine.AutomationEngineContext.Variables.Add(caseVal);
-            ScriptVariable case1 = new ScriptVariable();
-            case1.VariableName = "caseOutput";
-            case1.VariableValue = "unassigned";
-            _engine.AutomationEngineContext.Variables.Add(case1);
+            ScriptVariable caseInput = new ScriptVariable();
```
Comment (openbots-ff) [inline OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs:40]: You're not using the createtestvariable method here

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs:35]: You changed it in Task but not in Switch

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs:13]: You changed it in Task but not in Switch

**Review by openbots-ff — CHANGES_REQUESTED** [review]

File: OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs]: You also have script files in the resources though. Do you intend to keep those?

**Comment by openbots-ff** [comment]
Please resolve the merge conflicts

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/Properties/AssemblyInfo.cs]: Addressed as of 0271f65

File: OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Task.Test/Properties/AssemblyInfo.cs]: Removed unused resources as of 0271f65

File: OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs (hunk 5 above)
Comment (openbots-su) [reply OpenBots.Commands.Test/OpenBots.Commands.Switch.Test/SwitchCommandTests.cs]: Updated as of 0271f65

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**